src/pykoala/corrections/sky.py

```diff
@@ -71,443 +187,640 @@ def pol_continuum(data, wavelength, pol_order=3, **polfit_kwargs):
+
     Methods
     -------
-    substract
-    substract_PCA
+    subtract(data, variance, axis=-1)
+        Subtracts the sky model from the given data.
+    
+    subtract_pca()
+        Placeholder for PCA subtraction method.
+    
+    vprint(*messages)
+        Print messages if `verbose` is True.
     """
+
     verbose = True
 
     def __init__(self, **kwargs):
+        """
+        Initialize the SkyModel object.
+
+        Parameters
+        ----------
+        **kwargs : dict, optional
+            Dictionary of parameters to initialize the SkyModel.
+            Accepted keys are:
+            - wavelength : np.ndarray
+                1-D array representing the wavelengths of the sky model.
+            - intensity : np.ndarray
+                Array representing the intensity of the sky model.
+            - variance : np.ndarray
+                Array representing the variance associated with the sky model.
+            - verbose : bool
+                If True, print messages during execution. Default is True.
+        """
         self.wavelength = kwargs.get('wavelength', None)
         self.intensity = kwargs.get('intensity', None)
         self.variance = kwargs.get('variance', None)
         self.verbose = kwargs.get('verbose', True)
 
     def substract(self, data, variance, axis=-1, verbose=False):
-        """Substracts the sky_model to all fibres in the rss
+        """
+        Subtracts the sky model from the given data.
 
         Parameters
         ----------
-        data: (np.ndarray)
-            Data array for which the sky will be substracted
-        variance: (np.ndarray)
-            Array of variance data to include errors on determining the sky.
-        axis: (np.ndarray)
-            Spectral direction of data
+        data : np.ndarray
+            Data array from which the sky will be subtracted.
+        variance : np.ndarray
+            Array of variance data to include errors in determining the sky.
+        axis : int, optional
+            Spectral direction of the data. Default is -1.
+
         Returns
         -------
-        data_subs: (np.ndarray)
-        var_subs: (np.ndarray)
+        data_subs : np.ndarray
+            Data array after the sky model has been subtracted.
+        var_subs : np.ndarray
+            Variance array after including the sky model variance.
         """
-        # TODO
         if data.ndim == 3 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[:, np.newaxis, np.newaxis]
             skymodel_var = self.intensity[:, np.newaxis, np.newaxis]
         elif data.ndim == 2 and self.intensity.ndim == 1:
             skymodel_intensity = self.intensity[np.newaxis, :]
             skymodel_var = self.intensity[np.newaxis, :]
+        elif data.ndim == 2 and self.intensity.ndim == 2:  
+            skymodel_intensity = self.intensity       
+            skymodel_var = self.variance
```

Comment:
There must be a more elegant way to handle this (perhaps `np.expand_dims`?), but let's keep it for the time being. I had written a fairly long paragraph about the `RSS` and `Cube` classes, but I guess we should discuss those ideas as a separate issue.

Reply:
I fully agree. Where is that long paragraph you mention?

Reply:
I think I have more or less sorted out my ideas in <https://github.com/pykoala/koala/issues/439>